ex2_system/source/beacon/beacon_task.c

```diff
@@ -97,15 +39,28 @@ static void *beacon_daemon(void *pvParameters) {
         /* Constructing the system beacon content */
         // Refer to table 3 of MOP
         UHF_configStruct beacon_msg;
+        All_systems_housekeeping all_hk_data;
+
+        //Testing purposes only, passing 1s to the UHF
+        beacon_t beacon_packet = {
+            //define each element as 1 for testing purposes
+            1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1};
+
+        update_beacon(&all_hk_data);
+
+        memcpy(&(beacon_msg.message), &beacon_packet, sizeof(beacon_t));
+        uint8_t length = 97;
+        memcpy(&(beacon_msg.len), &length, sizeof(length));
         // TODO: call the appropriate HAL functions to get the most updated or
         // cached information of the components + state machine, RTC, etc.
         // Then uncomment the next line:
-        // uhf_status = HAL_UHF_setBeaconMsg(beacon_msg);
 
+        uhf_status = HAL_UHF_setBeaconMsg(beacon_msg);

```

Comment:
Make this function accept a pointer and pass the pointer to it instead of the struct